dependencies.yaml

```
@@ -58,19 +58,27 @@ dependencies:
           - matrix:
               cuda: "11.2"
             packages:
-              - cudatoolkit=11.2
+              - cuda-version=11.2
+              - cudatoolkit
           - matrix:
               cuda: "11.4"
             packages:
-              - cudatoolkit=11.4
+              - cuda-version=11.4
+              - cudatoolkit
           - matrix:
               cuda: "11.5"
             packages:
-              - cudatoolkit=11.5
+              - cuda-version=11.5
+              - cudatoolkit
           - matrix:
               cuda: "11.8"
             packages:
-              - cudatoolkit=11.8
+              - cuda-version=11.8
+              - cudatoolkit
+          - matrix:
+              cuda: "12.0"
+            packages:
+              - cuda-version=12.0
```

Think we may want these here analogous to [what `ucx` does for CUDA 12]( https://github.com/conda-forge/ucx-split-feedstock/blob/78e8d08b1391a6a29d05eb22bf8a30d1878954f2/recipe/meta.yaml#L37-L39 )

```suggestion
          - matrix:
              cuda: "12.0"
            packages:
              - cuda-version=12.0
              - cuda-cudart-dev
              - cuda-nvml-dev
```

---

Why? UCX-Py doesn't depend on any CUDA runtime dev packages to build. I think though that the intent here is to ensure they're available at runtime to properly test CUDA, is that right? If that is really the case then we may want those same packages but without `-dev`, WDYT?

---

This part of `dependencies.yaml` is used to generate package lists for development environments. While `ucx-py` doesn't have a developer conda environment in https://github.com/rapidsai/ucx-py/tree/branch-0.33/conda/environments, it might be nice to list any packages needed for building/testing here. I'm unfamiliar with how nvml is used in this package so I can't comment on the correctness of this suggestion.

---

Yeah that is fair. Was thinking this might be useful when building UCX-Py (and maybe also UCX) in an environment. That said, no strong feelings about this particular construction

If we want a strictly runtime focused solution, would suggest something like this

```suggestion
          - matrix:
              cuda: "12.0"
            packages:
              - cuda-version=12.0
              - cuda-cudart
```

---

Peter added the latter below in commit ( https://github.com/rapidsai/ucx-py/pull/967/commits/58d5ac29f3658a69f59a5784c839a540f91a248a )

Marking this resolved